src/output-filestore.c

```diff
@@ -0,0 +1,482 @@
+    snprintf(ctx->tmpdir, sizeof(ctx->tmpdir) - 1, "%s/tmp", log_directory);
+
+    OutputCtx *output_ctx = SCCalloc(1, sizeof(OutputCtx));
+    if (unlikely(output_ctx == NULL))
+        return NULL;
+
+    output_ctx->data = ctx;
+    output_ctx->DeInit = OutputFilestoreLogDeInitCtx;
+
+    const char *write_fileinfo = ConfNodeLookupChildValue(conf,
+            "write-fileinfo");
+    if (write_fileinfo != NULL && ConfValIsTrue(write_fileinfo)) {
+#ifdef HAVE_LIBJANSSON
+        SCLogConfig("File-store (v2) will output fileinfo records.");
+        ctx->fileinfo = true;
+#else
+        SCLogWarning(SC_ERR_NO_JSON_SUPPORT,
+                "Logging of fileinfo records requires JSON support.");
+#endif
+    }
+
+    const char *force_filestore = ConfNodeLookupChildValue(conf,
+            "force-filestore");
+    if (force_filestore != NULL && ConfValIsTrue(force_filestore)) {
+        FileForceFilestoreEnable();
+        SCLogInfo("forcing filestore of all files");
+    }
+
+    const char *force_magic = ConfNodeLookupChildValue(conf, "force-magic");
+    if (force_magic != NULL && ConfValIsTrue(force_magic)) {
+        FileForceMagicEnable();
+        SCLogInfo("forcing magic lookup for stored files");
+    }
+
+    FileForceHashParseCfg(conf);
+
+    /* The new filestore requires SHA256. */
+    FileForceSha256Enable();
+
+    const char *stream_depth_str = ConfNodeLookupChildValue(conf,
+            "stream-depth");
+    if (stream_depth_str != NULL && strcmp(stream_depth_str, "no")) {
+        uint32_t stream_depth = 0;
+        if (ParseSizeStringU32(stream_depth_str,
+                               &stream_depth) < 0) {
+            SCLogError(SC_ERR_SIZE_PARSE, "Error parsing "
+                       "file-store.stream-depth "
+                       "from conf file - %s.  Killing engine",
+                       stream_depth_str);
+            exit(EXIT_FAILURE);
+        } else {
+            FileReassemblyDepthEnable(stream_depth);
+        }
+    }
+
+    const char *file_count_str = ConfNodeLookupChildValue(conf,
+            "max-open-files");
+    if (file_count_str != NULL) {
+        uint32_t file_count = 0;
+        if (ParseSizeStringU32(file_count_str,
+                               &file_count) < 0) {
+            SCLogError(SC_ERR_SIZE_PARSE, "Error parsing "
+                       "file-store.max-open-files "
+                       "from conf file - %s.  Killing engine",
+                       stream_depth_str);
+            exit(EXIT_FAILURE);
+        } else {
+            if (file_count != 0) {
+                FileSetMaxOpenFiles(file_count);
+                SCLogInfo("file-store will keep a max of %d simultaneously"
+                          " open files", file_count);
+            }
+        }
+    }
+
+    SCReturnPtr(output_ctx, "OutputCtx");
+}
+
+#endif /* HAVE_NSS */
+
+void OutputFilestoreInitConfig(void)
+{
+#ifdef HAVE_NSS
+    StatsRegisterGlobalCounter("file_store.open_files",
```

Comment:
should this be v2 specific? In theory ppl can enable both v1 and v2 (although perhaps we should just block that from happening from the start)